src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -240,11 +240,10 @@ export default class QueryEditorUI extends Component<Props, State> {
 
     if (!isEqual(this.props.query.dataset, prevQuery.dataset)) {
       if (this.inputRef) {
-        const newQuery = this.queryString.getInitialQuery();
-        const newQueryString = newQuery.query;
-        if (this.inputRef.getValue() !== newQueryString) {
+        const newQueryString = this.props.query.query;
+        if (this.inputRef.getValue() !== newQueryString && typeof newQueryString === 'string') {
```

Comment:
is inputRef.getValue() behind prop because its a controlled input (e.g. its value is defined by props)?